### ISSUE TITLE
Add DNSSEC validation option

### ISSUE THREAD

**Issue by ameshkov** [AdguardTeam/dnsproxy#10]

**Comment by uBlock-user** [comment]
any plans for Encrypted SNI also ?

**Comment by ameshkov** [comment]
As a part of DOH implementation? Not until it is a part of Go language.

**Comment by uBlock-user** [comment]
Yes, does it have to server-sided or will having it on the client will do ?

**Comment by ameshkov** [comment]
It needs to be supported by both the client and the server. I think it's going to take years before ESNI is widely adopted considering that this is just a draft at the moment, and there is still some controversy about it.

**Comment by uBlock-user** [comment]
Firefox and Cloudflare already support it, so I hope to see it land in Chromium real soon as a feature request bug was opened back in October.

**Comment by ameshkov** [comment]
Well, yeah, and that's why I say years, otherwise I would say "never" :)

**Comment by Big--Mike** [comment]
Is there a roadmap or can you estimate, when DNSSEC support will be available?

**Comment by ameshkov** [comment]
No estimates yet. DNSSEC requires implementing a full-scale DNS recursor in go first.

**Comment by pemensik** [comment]
Hmm, nope. Dnsmasq is not full scale recursor and has dnssec validation. But it needs to do its own queries dependent on other queries. Not a simple feature. But please allow DNSSEC Pass through, meaning DO enabled query get DO enabled response, if upstream forwarders can provide it.

**Comment by ninetailedtori** [comment]
Bump
